dss-api.yml

```diff
@@ -176,10 +188,22 @@ paths:
               type: string
               pattern: "^[a-z0-9]{64}$"
     put:
-      summary: Create a file
+      summary: Create a new version of a file
       description: |
-        Create a new file with a given UUID.  The file content is passed in through a cloud URL.  The file on the cloud
-        provider must have metadata set reflecting the file checksums and the file content type.
+        Create a new version of a file with a given UUID.  The file content is passed in through a cloud URL.  The file
+        on the cloud provider must have metadata set reflecting the file checksums and the file content type.
+
+        The metadata fields required are:
+        - hca-dss-content-type: content-type of the file
+        - hca-dss-sha256: SHA-256 checksum of the file
+        - hca-dss-sha1: SHA-1 checksum of the file
+        - hca-dss-s3_etag: S3 ETAG checksum of the file.  See
+        https://stackoverflow.com/questions/12186993/what-is-the-algorithm-to-compute-the-amazon-s3-etag-for-a-file-larger-than-5gb
+        for how the general algorithm for how checksum is calculated.  For files smaller than 64MB, this is the MD5
+        checksum of the file.  For files larger than 64MB but smaller than 640GB, we use 64MB chunks.  For files larger
```

Comment:
I think for clarity about what "640GB" means, you may want to say "640000MB" instead. I believe you want `1024*1024*64*10000` as the cutoff, not `1024*1024*1024*640`. If we select the former and the client selects the latter, the part sizes are within bounds, but the etags won't match.